packages/astro/src/content/utils.ts

```diff
@@ -49,6 +49,14 @@ export const msg = {
 		`${collection} does not have a config. We suggest adding one for type safety!`,
 };
 
+export function extractFrontmatterAssets(data: Record<string, any>): string[] {
+	const assets = Object.values(data).filter(
+		(value) => typeof value === 'object' && value['__astro_asset'] == true
+	);
```

Comment:
Looks like this only supports `image()` at the top level. Does this need to be recursive?